From d41d57dcb15694b04bf0022a62b0867ead94e807 Mon Sep 17 00:00:00 2001
From: Matt DeVillier <matt.devillier@puri.sm>
Date: Thu, 17 Jun 2021 18:54:16 -0500
Subject: [PATCH 3/4] mb/purism/librem_14/hda_verb.c: Fix subsystem, verb
 count, jack detect

- set subsystem/subvendor ID to Realtek default, as the one dumped
  from the vendor UEFI firmware provides no advantages
- fix the number of verb entries, which excluded the 4 following the
  pin configs
- issue the reset *before* setting the subvendor, and use the Azalia macro
- disable jack detect for the 3.5mm jack, both line out and mic, since
  it's not currently working, so that the outputs can be manually selected

Change-Id: Icd961c3c5aec23cf61d6a9ad65c603c6dc04697a
Signed-off-by: Matt DeVillier <matt.devillier@puri.sm>
---
 .../librem_cnl/variants/librem_14/hda_verb.c    | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/src/mainboard/purism/librem_cnl/variants/librem_14/hda_verb.c b/src/mainboard/purism/librem_cnl/variants/librem_14/hda_verb.c
index 97422ef317..1256c1721b 100644
--- a/src/mainboard/purism/librem_cnl/variants/librem_14/hda_verb.c
+++ b/src/mainboard/purism/librem_cnl/variants/librem_14/hda_verb.c
@@ -4,27 +4,22 @@
 
 const u32 cim_verb_data[] = {
 	0x10ec0256,	/* Codec Vendor/Device ID: Realtek ALC256 */
-	0x1b50580a,	/* Subsystem ID */
-	12,		/* Number of entries */
+	0x10ec0256,	/* Subsystem ID */
+	16,		/* Number of entries */
 
-	AZALIA_SUBVENDOR(0, 0x1b50580a),
-
-	/* Widget node 0x01 */
-	0x0017ff00,
-	0x0017ff00,
-	0x0017ff00,
-	0x0017ff00,
+	AZALIA_RESET(0x1),
 
+	AZALIA_SUBVENDOR(0, 0x10ec0256),
 	AZALIA_PIN_CFG(0, 0x12, 0x90a60140), /* Front digital mic */
 	AZALIA_PIN_CFG(0, 0x13, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x14, 0x90170110), /* Internal speakers */
 	AZALIA_PIN_CFG(0, 0x18, 0x411111f0), /* NC */
-	AZALIA_PIN_CFG(0, 0x19, 0x04a11030), /* Jack analog mic */
+	AZALIA_PIN_CFG(0, 0x19, 0x04a11130), /* Jack analog mic */
 	AZALIA_PIN_CFG(0, 0x1a, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x1b, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x1d, 0x411111f0), /* NC */
 	AZALIA_PIN_CFG(0, 0x1e, 0x411111f0), /* NC */
-	AZALIA_PIN_CFG(0, 0x21, 0x04211020), /* Jack analog out */
+	AZALIA_PIN_CFG(0, 0x21, 0x04211120), /* Jack analog out */
 
 	/* Hidden SW reset */
 	0x0205001a,
-- 
2.30.2

